src/ServiceProvider.php

```diff
@@ -15,23 +20,30 @@ class ServiceProvider extends LaravelServiceProvider
     protected $defer = false;
 
     /**
-     * Bootstrap the application events.
+     * Register the service provider (for bindings).
      *
      * @return void
      */
-    public function boot()
+    public function register()
     {
-        $this->config();
-        $this->configureCommands();
+        $this->improveGettextFunctions();
     }
 
     /**
-     * Register the service provider.
+     * Bootstrap the application events (executed after all services are booted).
      *
      * @return void
      */
-    public function register()
+    public function boot()
     {
+        $this->config();
+        $this->configureCommands();
+
+        $config = config('translationio');
+
+        $translationIO = new TranslationIO($config);
+
+        $translationIO->setLocale($config['source_locale']);
```

Comment:
!! Dont think we can do this here, this should be done by user, depending on their add logic

Reply:
We did it like this in Rails and it doesn't seem to bother anyone. You always need a default language and the source locale seems a good fit.

In the code, the best way to manage locales is to add setLocale at controller's level (using session, cookie, params in URL, subdomain or anything else).

In Rails this is the code we suggest to users and it will always override the default locale: https://github.com/aurels/translation-gem/blob/8ccb7df8b39248d2f408d93d67ee72651f7bccbf/lib/translation_io/controller.rb#L3

Reply:
You are right, cant explain why I though it could cause any problems..